pkg/validator.go

```diff
@@ -0,0 +1,74 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+)
+
+
+type QuotaValidator struct {
+	RulesContainer *RulesContainer
+}
+
+func (qv *QuotaValidator) Handle(quota string,history OrderHistory, newOrder string ) error {
+	//todo: move to map number of products increase
+	switch quota {
+		case BASIC: return qv.onBasic(history,newOrder)
+		case MANIAC: return qv.onManiac(history,newOrder)
+		case LOVER: return qv.onLover(history,newOrder)
+	}
+	
+	return errors.New("no such quota: " + quota)
+}
+
+func (qv *QuotaValidator) onBasic(history OrderHistory, newOrder string) error {
+	rules := qv.RulesContainer.getRule(BASIC)
+
+	return qv.validate(history, newOrder, rules ,24)
+}
+
+func (qv *QuotaValidator) onLover(history OrderHistory,newOrder string) error {
+	rules := qv.RulesContainer.getRule(LOVER)
+
+	return qv.validate(history, newOrder, rules ,24)
+}
+
+func (qv *QuotaValidator) onManiac(history OrderHistory, newOrder string) error {
+	rules := qv.RulesContainer.getRule(MANIAC)
+	var timeWindow int 
+	if newOrder == ESPRESSO {
```

Comment:
ah, here it is. This is a hack. This code tighten to much to actually configuration.
Changing the quoats should be done in one place, e.g. changing the `config.go` (or `config.yaml`) - so one place where you store the config. But not in some random functions

Reply:
fixed